File: 1647-minimum-deletions-to-make-character-frequencies-unique/1647-minimum-deletions-to-make-character-frequencies-unique.py
class Solution:
    def minDeletions(self, s: str) -> int:
        freq = [0]*26
        for ch in s:
            freq[ord(ch) - 97] += 1
        
        # print(freq)
        
        freq.sort(reverse=True)
        cnt = 0
        maxi = freq[0]
        for i in range(1,26):
            if freq[i] == 0:
                continue
            if freq[i] < maxi:
                maxi = freq[i]
            else:
                maxi = max(0,maxi-1)
                cnt += freq[i] - maxi
                freq[i] = maxi
                
        return cnt
        
        
        

File: 1647-minimum-deletions-to-make-character-frequencies-unique/1647-minimum-deletions-to-make-character-frequencies-unique.cpp
class Solution {
public:
    int minDeletions(string s) {
        int cnt = 0;
        vector<int> freq(26);
        for(auto &ch : s)
            freq[ch - 'a']++;
        sort(freq.begin(),freq.end(),greater<int>());
        
        int maxi = freq[0];
        for(int i=1;i<26;i++)
        {
            if(freq[i] == 0)
                break;
            if(freq[i] < maxi) maxi = freq[i];
            else maxi = max(0,maxi-1) , cnt += freq[i] - maxi;
        }
        
        return cnt;
        
    }
};